packages/simcore-sdk/src/simcore_sdk/node_ports_common/r_clone.py

```diff
@@ -0,0 +1,134 @@
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.STDOUT,
+        cwd=cwd,
+    )
+
+    stdout, _ = await proc.communicate()
+    decoded_stdout = stdout.decode()
+    if proc.returncode != 0:
+        raise _CommandFailedException(command=str_cmd, stdout=decoded_stdout)
+
+    logger.debug("'%s' result:\n%s", str_cmd, decoded_stdout)
+    return decoded_stdout
+
+
+@cached()
+async def is_r_clone_available(r_clone_settings: Optional[RCloneSettings]) -> bool:
+    """returns: True if the `rclone` cli is installed and a configuration is provided"""
+    try:
+        await _async_command("rclone", "--version")
+        return r_clone_settings is not None
+    except _CommandFailedException:
+        return False
+
+
+async def sync_local_to_s3(
+    session: ClientSession,
+    r_clone_settings: RCloneSettings,
+    s3_object: str,
+    local_file_path: Path,
+    user_id: UserID,
+) -> ETag:
+
+    s3_link = await get_upload_file_presigned_link(
+        session=session,
+        file_id=s3_object,
+        location_id="0",  # only works with simcore s3
+        user_id=user_id,
+        as_presigned_link=False,
+    )
+    s3_path = re.sub(r"^s3://", "", s3_link)
+    logger.debug(" %s; %s", f"{s3_link=}", f"{s3_path=}")
+
+    r_clone_config_file_content = get_r_clone_config(r_clone_settings)
+    async with _config_file(r_clone_config_file_content) as config_file_name:
+        source_path = local_file_path
+        destination_path = Path(s3_path)
+        file_name = local_file_path.name
+        # TODO: capture and send progress somehow?
+
+        # rclone only acts upon directories, so to target a specific file
+        # we must run the command from the file's directory. See below
+        # example for further details:
+        #
+        # local_file_path=`/tmp/pytest-of-silenthk/pytest-80/test_sync_local_to_s30/filee3e70682-c209-4cac-a29f-6fbed82c07cd.txt`
+        # s3_path=`simcore/00000000-0000-0000-0000-000000000001/00000000-0000-0000-0000-000000000002/filee3e70682-c209-4cac-a29f-6fbed82c07cd.txt`
+        #
+        # rclone
+        #   --config
+        #   /tmp/tmpd_1rtmss
+        #   sync
+        #   '/tmp/pytest-of-silenthk/pytest-80/test_sync_local_to_s30'
+        #   'dst:simcore/00000000-0000-0000-0000-000000000001/00000000-0000-0000-0000-000000000002'
+        #   --progress
+        #   --copy-links
+        #   --include
+        #   'filee3e70682-c209-4cac-a29f-6fbed82c07cd.txt'
+        r_clone_command = (
+            "rclone",
+            "--config",
+            config_file_name,
+            "sync",
+            f"'{source_path.parent}'",
+            f"'dst:{destination_path.parent}'",
+            "--progress",
+            "--copy-links",
+            "--include",
+            f"'{file_name}'",
+        )
+
+        try:
+            await _async_command(*r_clone_command, cwd=f"{source_path.parent}")
+            return await update_file_meta_data(
```

Comment:
Q: Why is this fct touching the database at all? It says sync_with_s3 not sync_with_everything.